### PR TITLE
Refs [TASK][44582] coin details data source

### PR DESCRIPTION
**Pull request by ghost**
Related Tickets
#Task 44582: coin details data source
WHAT
Evidence (Screenshot or Video)
Review Checklist
Category	View Point	Description	Self review	Reviewer2 (name)
Conventions	Sun* coding conventions	https://github.com/framgia/coding-standards/tree/master/eng/android	
 yes
 yes
Conventions	Kotlin coding conventions	https://kotlinlang.org/docs/coding-conventions.html	
 yes
 yes
Redmine	Sun* Redmine working process	https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md	
 yes
 yes
Notes (Optional)
(Impacted Areas in Application(List features, api, models or services that this PR will affect))

(List gem, library third party add new)

(Other notes)

### CODE REVIEW

File: lib/bloc/coin_details/coin_details_bloc.dart [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:interactive_chart/interactive_chart.dart';
+import 'package:vn_crypto/bloc/coin_details/coin_details_event.dart';
+import 'package:vn_crypto/bloc/coin_details/coin_details_state.dart';
+import 'package:vn_crypto/data/model/coin_detail.dart';
+import 'package:vn_crypto/data/repository/coins_repository.dart';
+import 'package:vn_crypto/ultils/Constant.dart';
+
+
```
Comment (tuannm-1276) [inline lib/bloc/coin_details/coin_details_bloc.dart]: bl

File: lib/bloc/coin_details/coin_details_bloc.dart [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  }
+
+  void _onGetCoin(var event, var emit) async {
+    String coinId = (event as CoinDetailsLoaded).coinId;
+    emit(CoinDetailsLoading());
+    CoinDetails coin =
+        await coinRepository.getCoin(coinId);
+    List<CandleData> candleDatas = await coinRepository.getCoinOhlc(coinId, AppStrings.textCurrency, Constant.defaultDays);
+    if (coin == null) {
+      emit(CoinDetailsLoadFailed(AppStrings.errorLoadDataFailed));
```
Comment (vietnh-1109) [inline lib/bloc/coin_details/coin_details_bloc.dart:24]: cái này ko check empty hở Nam

File: lib/bloc/coin_details/coin_details_state.dart [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+class CoinDetailsLoading extends CoinDetailsState {}
+
+class CoinDetailsLoadSuccess extends CoinDetailsState {
+  final CoinDetails coin;
+  final List<CandleData> candleDatas;
+  CoinDetailsLoadSuccess({required this.coin, required this.candleDatas});
+
+  @override
+  List<Object> get props => [coin];
```
Comment (vietnh-1109) [inline lib/bloc/coin_details/coin_details_state.dart]:
```suggestion
  @override
  List<Object> get props => [coin , candleData];
```

File: lib/bloc/coin_details/coin_details_bloc.dart (hunk 2 above)
Comment (ghost) [reply lib/bloc/coin_details/coin_details_bloc.dart:24]: đây là object thôi không phải List Việt ơi

**Review by vietnh-1109 — APPROVED** [review]

**Pull request merged**